### PR TITLE
Add Markdown again

### PR DESCRIPTION
**Pull request by theHamsta**
I will test this for a while...

### CODE REVIEW

**Comment by theHamsta** [comment]
If someone has time, they can do some fuzzing of the new verison...

File: lua/nvim-treesitter/parsers.lua [hunk 1]
```diff
@@ -295,13 +295,12 @@ list.haskell = {
...
--- list.markdown = {
---   install_info = {
---     url = "https://github.com/ikatyang/tree-sitter-markdown",
---     files = { "src/parser.c", "src/scanner.cc" },
---   }
--- }
+ list.markdown = {
+   install_info = {
+     url = "https://github.com/ikatyang/tree-sitter-markdown",
+     files = { "src/parser.c", "src/scanner.cc", "-DTREE_SITTER_MARKDOWN_AVOID_CRASH=1" },
```
Comment (theHamsta) [inline lua/nvim-treesitter/parsers.lua:301]: wow, this works even for `cl.exe`

**Comment by tami5** [comment]
Thanks a lot @theHamsta I'll test it out in an hour

**Comment by theHamsta** [comment]
Best would be to fuzz the parser. I still don't trust it. But using it a week or so might be useful also.

**Comment by tami5** [comment]
> Best would be to fuzz the parser. I still don't trust it. But using it a week or so might be useful also.

Okay how do I fuzz the parser :$ :D

**Comment by theHamsta** [comment]
You can read an instruction in the original SEGFAULT issue. You need to find out how to compile the parser in the test ASAN script with exception and this AVOID_CRASH definition

**Comment by theHamsta** [comment]
I had one situation where i might had a infinite loop when I had a markdown file open. This might also had a different cause...

**Comment by aktau** [comment]
Especially since you have a repro case for such an incredibly long parse, it'd be good to file this as an issue to https://github.com/ikatyang/tree-sitter-markdown (cc @ikatyang).

**Comment by theHamsta** [comment]
@aktau I can't reproduce with the current nvim. Maybe there was something wrong with a former nvim revision.

We still don't know whether this parser is reliable. Did you do some testing or fuzzing?

**Comment by theHamsta** [comment]
Having the infinite loop issue with another README file. I guess the simplest solution would be to just take the external rules of the HTML scanner and start with a new grammar.js (maybe with indent/dedent from Python). At least it would be impossible for me to identify any problems with the current implementation.

**Comment by dimbleby** [comment]
https://github.com/ikatyang/tree-sitter-markdown/issues/32

**Comment by theHamsta** [comment]
At the moment we don't have any plans for this

**Pull request closed without merging**